lib/src/state_manager/rx/rx_workers/rx_workers.dart

```diff
@@ -1,115 +1,195 @@
+///
+/// Sample:
+/// Every time increment() is called, ever() will process the [condition]
+/// (can be a [bool] expression or a [bool Function()]), and only call the callback
+/// when [condition] is true.
+/// In our case, only when count is bigger to 5. In order to "dispose" this Worker
+/// that will run forever, we made a [worker] variable. So, when the count value
+/// reaches 10, the worker gets disposed, and releases any memory resources.
+///
+/// ```
+/// // imagine some counter widget...
+///
+/// class _CountController extends GetxController {
+///   final count = 0.obs;
+///   Worker worker;
+///
+///   void onInit() {
+///     worker = ever(count, (value) {
+///       print('counter changed to: $value');
+///       if (value == 10) worker.dispose();
+///     }, condition: () => count > 5);
+///   }
+///
+///   void increment() => count + 1;
+/// }
+/// ```
 Worker ever<T>(RxInterface<T> listener, Function(T) callback,
-    {bool condition = true}) {
+    {dynamic condition = true}) {
   StreamSubscription sub = listener.subject.stream.listen((event) {
-    if (condition) callback(event);
+    if (_conditional(condition)) callback(event);
   });
-
-  Future<void> cancel() {
-    return sub.cancel();
-  }
-
-  return Worker(cancel, '[ever]');
+  return Worker(sub.cancel, '[ever]');
 }
 
-Worker everAll(List<RxInterface> listener, Function(dynamic) callback,
-    {bool condition = true}) {
+/// Similar to [ever], but takes a list of [listeners], the condition for the [callback]
+/// is common to all [listeners], and the [callback] is executed to each one of them.
+/// The [Worker] is common to all, so [worker.dispose()] will cancel all streams.
+Worker everAll(List<RxInterface> listeners, Function(dynamic) callback,
+    {dynamic condition = true}) {
   List<StreamSubscription> evers = <StreamSubscription>[];
-
-  for (var i in listener) {
+  for (var i in listeners) {
     StreamSubscription sub = i.subject.stream.listen((event) {
-      if (condition) callback(event);
+      if (_conditional(condition)) callback(event);
     });
     evers.add(sub);
   }
 
   Future<void> cancel() {
-    for (var i in evers) {
-      i.cancel();
-    }
+    for (var i in evers) i.cancel();
     return Future.value(() {});
   }
 
   return Worker(cancel, '[everAll]');
 }
 
+/// [once()] will execute only 1 time when [condition] is met and cancel
+/// the subscription to the [listener] stream right after that.
+/// [condition] defines when [callback] is called, and
+/// can be a [bool] or a [bool Function()].
+///
+/// Sample:
+/// ```
+///  class _CountController extends GetxController {
+///   final count = 0.obs;
+///   Worker worker;
+///
+///   @override
+///   Future<void> onInit() async {
+///     worker = once(count, (value) {
+///       print("counter reached $value before 3 seconds.");
+///     }, condition: () => count() > 2);
+///     3.delay(worker.dispose);
```

Comment:
What is this syntax that i don't know?
dart has that by default?

Reply:
He added the duration and delay extension in the last update, it is not from dart, but it is included in Getx

Reply:
Is just a utility for `num`. I've got tired of `await Future.delayed(Duration( seconds: 1 ));`, so the extension go for `num` and `Duration` (although, obviously you can't make it `const`) :

Duration:
`await 2.minutes.delay()` = `await 0.5.seconds.delay()`.
`DateTime.now().add(4.days);`

num: (the `delay()` always resolves as "seconds").
`await 0.3.delay();`